meadow-client/src/Marlowe/Semantics.purs

```diff
@@ -45,19 +48,45 @@ derive instance ordIdInput :: Ord IdInput
 type TimeoutData
   = M.Map Timeout (S.Set IdCommit)
 
-type CommitInfoRecord
-  = {person :: Person, amount :: BigInteger, timeout :: Timeout}
+newtype CommitInfoRecord
+  = CommitInfoRecord {person :: Person, amount :: BigInteger, timeout :: Timeout}
 
-data CommitInfo
+derive instance newtypeCommitInfoRecord :: Newtype CommitInfoRecord _
+instance eqCommitInfoRecord :: Eq CommitInfoRecord where
+  eq a b = equal (unwrap a) (unwrap b)
+derive instance genericCommitInfoRecord :: Generic CommitInfoRecord _
+instance showCommitInfoRecord :: Show CommitInfoRecord where
+  show = genericShow
+
+newtype CommitInfo
   = CommitInfo {redeemedPerPerson :: M.Map Person BigInteger, currentCommitsById :: M.Map IdCommit CommitInfoRecord, expiredCommitIds :: S.Set IdCommit, timeoutData :: TimeoutData}
 
-type OracleDataPoint
-  = {blockNumber :: BlockNumber, value :: BigInteger}
+derive instance newtypeCommitInfo :: Newtype CommitInfo _
+instance eqCommitInfo :: Eq CommitInfo where
+  eq a b = equal (unwrap a) (unwrap b)
```

Comment:
I wonder if the Prelude's `on` function is worth using here?

```
eq = equal `on` unwrap
```

https://pursuit.purescript.org/packages/purescript-prelude/4.1.1/docs/Data.Function#v:on

Reply:
This comment applies in a few places down below too.

Reply:
nice

Reply:
done